semgrep-core/src/parsing/tree_sitter/Parse_hack_tree_sitter.ml

```diff
@@ -2859,11 +2864,19 @@ let parse file =
         H.debug_sexp_cst_after_error (CST.sexp_of_script cst);
         raise exn)
 
+let parse_expression_or_source_file str =
+  let res = Tree_sitter_hack.Parse.string str in
+  match res.errors with
+  | [] -> res
+  | _ ->
+      let expr_str = str ^ ";" in
+      Tree_sitter_hack.Parse.string expr_str
```

Comment:
Here I'd say "whatever works". The `__SEMGREP_EXPRESSION` hack is pretty ugly anyway.